assets/app.js

```diff
@@ -7,11 +7,11 @@ const $ = require('jquery');
 $(function () {
     // Initiate Select2, which allows dynamic entry of languages.
     $('#lang').select2({
-        tags: true,
         theme: 'bootstrap',
     });
 
     // Show engine-specific options.
+    // TODO: Update the Select2 options with available languages.
```

Comment:
This I think would normally be a big problem for usability, but I understand that's less important since most people will never see it. I imagine we might be redesigning everything anyway, so I didn't want to put too much work into the user-facing stuff. For now you can force it to use the Tesseract list by putting ?engine=tesseract in the URL.

Reply:
Also, I assume we'll want to map TesseractEngine::getValidLangs() to return ISO 639-1 for consistency with what people are used to on-wiki, and also so that Intuitions `getLangName()` will work. Right now with Tesseract's format you don't get the language name in the autocompletion.

Reply:
You do get the language name for some languages, although I'm not really sure why. I've commented on the ticket https://phabricator.wikimedia.org/T282073#7064604